### ISSUE TITLE
Raise map limit

### ISSUE THREAD

**Issue by Jaffies** [Facepunch/garrysmod-requests#2130]
https://github.com/nillerusr/source-engine/commit/9b1d5a0575fa8ec0519b82b16bd4f800b96093cd
litteraly does it.

**Comment by DarkHoshipoN** [comment]
GMod definitely needs this.

**Comment by FrantakPlay** [comment]
This could solve a lot of problems

**Comment by FlorianLeChat** [comment]
https://github.com/Facepunch/garrysmod-requests/issues/448#issuecomment-91297808

**Comment by Mingooberus** [comment]
> [#448 (comment)](https://github.com/Facepunch/garrysmod-requests/issues/448#issuecomment-91297808)

Lack of foresight, maybe don't increase the grid to Sourcebox limits which is around 50 miles, but maybe to Contagion levels?

**Comment by Jaffies** [comment]
Imagine this in Gmod
https://youtu.be/3Ka9gKkHDTM

**Comment by SpiffyJUNIOR** [comment]
More info here: https://github.com/Facepunch/garrysmod-requests/issues/448#issuecomment-152797392

I think this is possible, I expected it to be far buggier than the comment above described.

**Comment by noaccessl** [comment]
The SourceBox repository has been deleted/hidden, so I decided to gather all the commits related to expanding the map size limit here. Maybe Rubat will take on this task, and I think it will be easier for him this way.
By the way, why not create a separate branch for this? So that if something goes wrong, it doesn't affect the majority of servers and players.

https://github.com/nillerusr/source-engine/commit/9b1d5a0575fa8ec0519b82b16bd4f800b96093cd - `LMAOOOOOOOO` (first)
https://github.com/nillerusr/source-engine/commit/296b4316537e502fb8675323ce152646b57a4a45 - `voxelBounds now authomatically changes with the COORD_INTEGER_BITS`
https://github.com/nillerusr/source-engine/commit/d9a48ea6dda0658484c40a7318f9329b4c4e53d3 - `epic`
https://github.com/nillerusr/source-engine/commit/12c6c2157ae201a0c361d2641df6b41f6dd9c476 - `🦀WE BROKE COMPATIBLITY WITH EVERY SOURCE GAME 🦀`
https://github.com/nillerusr/source-engine/commit/130dde7102b8012404e735e8ef5302a16acdc97d - `🦀WE ARE NOW COMPATIBLE WITH VALVE GAMES🦀` - `AND ALSO HAVE VERY BIG MAPS WORKING`
https://github.com/nillerusr/source-engine/commit/7c05c9998f5bfcd82ba0df197eb0c465567c8b61 - `i hate`
https://github.com/nillerusr/source-engine/commit/cc8f0f4141d9df38082b4d2f1285888be2f0cc45 - `fixed circles`
https://github.com/nillerusr/source-engine/commit/25e30be665bcb20cf4c000e00d7de72e4fd8c657 - `blocks_size option for vbsp`
https://github.com/nillerusr/source-engine/commit/a00bdff5d238c22a6b27a91d95f004c145526ed0 - `voice fixed`
https://github.com/nillerusr/source-engine/commit/79e3c55465bcb172362ed254ffb9dc0583569872 - `EPIC FIXES!!!`

**Comment by Jaffies** [comment]
> The SourceBox repository has been deleted/hidden, so I decided to gather all the commits related to expanding the map size limit here. Maybe Rubat will take on this task, and I think it will be easier for him this way. By the way, why not create a separate branch for this? So that if something goes wrong, it doesn't affect the majority of servers and players.
> 
> [nillerusr/source-engine@9b1d5a0](https://github.com/nillerusr/source-engine/commit/9b1d5a0575fa8ec0519b82b16bd4f800b96093cd) - `LMAOOOOOOOO` (first) [nillerusr/source-engine@296b431](https://github.com/nillerusr/source-engine/commit/296b4316537e502fb8675323ce152646b57a4a45) - `voxelBounds now authomatically changes with the COORD_INTEGER_BITS` [nillerusr/source-engine@d9a48ea](https://github.com/nillerusr/source-engine/commit/d9a48ea6dda0658484c40a7318f9329b4c4e53d3) - `epic` [nillerusr/source-engine@12c6c21](https://github.com/nillerusr/source-engine/commit/12c6c2157ae201a0c361d2641df6b41f6dd9c476) - `🦀WE BROKE COMPATIBLITY WITH EVERY SOURCE GAME 🦀` [nillerusr/source-engine@130dde7](https://github.com/nillerusr/source-engine/commit/130dde7102b8012404e735e8ef5302a16acdc97d) - `🦀WE ARE NOW COMPATIBLE WITH VALVE GAMES🦀` - `AND ALSO HAVE VERY BIG MAPS WORKING` [nillerusr/source-engine@7c05c99](https://github.com/nillerusr/source-engine/commit/7c05c9998f5bfcd82ba0df197eb0c465567c8b61) - `i hate` [nillerusr/source-engine@cc8f0f4](https://github.com/nillerusr/source-engine/commit/cc8f0f4141d9df38082b4d2f1285888be2f0cc45) - `fixed circles` [nillerusr/source-engine@25e30be](https://github.com/nillerusr/source-engine/commit/25e30be665bcb20cf4c000e00d7de72e4fd8c657) - `blocks_size option for vbsp` [nillerusr/source-engine@a00bdff](https://github.com/nillerusr/source-engine/commit/a00bdff5d238c22a6b27a91d95f004c145526ed0) - `voice fixed` [nillerusr/source-engine@79e3c55](https://github.com/nillerusr/source-engine/commit/79e3c55465bcb172362ed254ffb9dc0583569872) - `EPIC FIXES!!!`
You should have added "RUBAT PLEASE MAKE IT HAPPEN" in the end of your comment ;)

**Comment by ThomasMartin997** [comment]
This is a wonderful thing, hopefully it happens! It was always a lot of peoples dream to see this type of thing done in garry's mod

**Comment by Nalo12** [comment]
![image](https://github.com/Facepunch/garrysmod-requests/assets/89763856/02a89d60-fd90-4add-b8b1-8d0e3f87d8db)

**Comment by noaccessl** [comment]
Request is still open, so who knows?. ~~Rubat was in a hurry, so he just forgot to append "t now" to "no"~~

#2253 was closed just due to being a duplicate of this one.

> ..., so I have decided to gather all the commits related to expanding the map size limit here. **Perhaps** Rubat will take on this task, and I believe it will be easier for him this way.

**Comment by rytp-tv** [comment]
I agree. Limits are the only thing that prevents Gmod from becoming definitively great, giving rise to a new era of aircraft, rail and ship modes.

**Comment by Acer202** [comment]
> I agree. Map limits are the only one thing that prevents Gmod from becoming definitively great, giving rise to a new era of aircraft, rail and ship modes.

I'm a huge plane building nerd in Garrysmod and I've been crossing my fingers that they'll find a way to expand map sizes for about 15 years now, hah. I'm not counting on it—I don't know the technical side of things that's prevented it from happening already, but I agree with others that it would be a dream come true. I'm aware of the infinite map concept but those are very limited in form and function compared to just a normal map, but larger.

**Comment by pixel844** [comment]
+1 would be an awesome feature

**Comment by Sectonid** [comment]
Rubat please stop being quiet, give us a final response on this feature request: will you work on it or not? I know you have the request here as old as [June 7th, 2013](https://github.com/Facepunch/garrysmod-requests/issues/13), but this does not deserve the radio silence.

**Comment by pixel844** [comment]
hype and goodbye

**Comment by rytp-tv** [comment]
> > I agree. Map limits are the only one thing that prevents Gmod from becoming definitively great, giving rise to a new era of aircraft, rail and ship modes.
> 
> I'm a huge plane building nerd in Garrysmod and I've been crossing my fingers that they'll find a way to expand map sizes for about 15 years now, hah. I'm not counting on it—I don't know the technical side of things that's prevented it from happening already, but I agree with others that it would be a dream come true. I'm aware of the infinite map concept but those are very limited in form and function compared to just a normal map, but larger.

So far, you can test my **NoMapBorders** add-on, it does about the same thing as Infinity Maps, but it can be embedded and run on almost any map. It still needs some edits, like checking for "stuck in walls," but it's more or less working. I'm not sure if it should be developed, given that the official Gmod developers can expand the size limits of the maps at any time and there will be no need for such an addition.

**Comment by trainchasewastaken** [comment]
It was, in fact, not coming anytime soon.
